### DISCUSSION TITLE
$filters in Front-end API

### DISCUSSION THREAD

**Discussion by Kealtie** [swellstores/community#186] [Help] [answered]
Hello, I am currently developing a product category overview page with a list of products that are connected to a category.

I am doing this in a Nuxt 3 application. 

I came to the conclusion whilst implementing $filters, that some attributes that have the Number or Text type don't seem to give any results, even if the value is right. Can somebody explain to me what I am doing wrong here?

`const data = await swell.products.list({
            sort      : this.getSortingByOptions(),
            expand    : ['categories', 'variants'],
            categories: categoryIds,
            limit     : 2,
            page      : this.page,
            $filters  : this.activeFilters
        });`

the this.activeFilters in this particular case is an object that looks like this
`{
     length: ['600']
}`

I've also tried to map it to contain number type like so:
`{
      length: [600]
}`

When this.activeFilters contain filterable attributes like brand that has the 'select' type, it does work; which looks like this in my case:
`{
      brand: ['Rockfon']
}`

Would love to hear from you.

**Comment by Kealtie** [comment]
Also, 

`const filters = await swell.products.filters(data);`

gives non filterable attributes back (untoggled the filterable option in Swell). And doesn't give all the options for attributes back if you have a paginated product list fetch, isn't it currently quite useless this way?

**Comment by sags95** [comment, marked as answer]
@jordytenden we are currently investigating an issue with swell-js when it comes to filtering with number attributes. As a workaround in the meantime you can set the attribute type as text, you'll still be able to input the value as a number within the field (it will then be encoded as a string).

**Reply by Kealtie** [reply]
Alright, when can we expect a fix?

**Reply by Kealtie** [reply]
Also, changing the type to 'Text' doesn't actually change the Type of the value in the response of swellJS. I have to remove all the current values and re-enter them for each product.

**Comment by steven-k1** [comment]
@Kealtie A fix for the raised issue with number attribute filtering has been deployed. We recommend updating to the latest version of Swell.js for this to be reflected.
